### DISCUSSION TITLE
TypeScript keeps complaining about `Property 'as' is missing in type`

### DISCUSSION THREAD

**Discussion by iamtekeste** [react-hook-form/react-hook-form#2292] [Q&A] [answered]
As the title says even though I used `render` it keeps complaining about `as` being missing.

```js
<Controller
        control={control}
        name="test"
        render={({ onChange, onBlur, value }) => (
          <Input onChange={(e) => onChange(e.detail.value)} value={value} />
        )}
  />
```

What am I doing wrong here?

**Edit** 
If it helps I am using version react-hook-form@^5.7.2

**Comment by bluebill1049** [comment, marked as answer]
render is for v6 :(

**Comment by iamtekeste** [comment]
Yup that was the problem, thank you for your help. I was using the wrong version.

**Reply by bluebill1049** [reply]
np :)
